zerver/lib/export.py

```diff
@@ -1680,25 +1680,33 @@ def percent_callback(complete: Any, total: Any) -> None:
         sys.stdout.write('.')
         sys.stdout.flush()
 
+    public_url = ''  # type: str
     if settings.LOCAL_UPLOADS_DIR is not None:
-        raise CommandError("S3 backend must be configured to upload to S3")
-
-    print("Uploading export tarball to S3")
-
-    from zerver.lib.upload import S3Connection, get_bucket, Key
-    conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
-    # We use the avatar bucket, because it's world-readable.
-    bucket = get_bucket(conn, settings.S3_AVATAR_BUCKET)
-    key = Key(bucket)
-    key.key = os.path.join("exports", generate_random_token(32), os.path.basename(tarball_path))
-    key.set_contents_from_filename(tarball_path, cb=percent_callback, num_cb=40)
-
-    public_url = 'https://{bucket}.{host}/{key}'.format(
-        host=conn.server_name(),
-        bucket=bucket.name,
-        key=key.key)
-    print()
-    print("Uploaded to %s" % (public_url,))
+        path = "/".join([
+            'exports',
+            str(realm.id),
+            random_name(18)
+        ])
+        abs_path = settings.LOCAL_UPLOADS_DIR + path
+        os.makedirs(abs_path)
+        shutil.copy(tarball_path, abs_path)
+        print("Successfully uploaded the tarball to %s" % (abs_path,))
+        public_url = '/public/' + path
+    else:
+        print("Uploading export tarball to S3")
+        conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
+        # We use the avatar bucket, because it's world-readable.
+        bucket = get_bucket(conn, settings.S3_AVATAR_BUCKET)
+        key = Key(bucket)
+        key.key = os.path.join("exports", generate_random_token(32), os.path.basename(tarball_path))
+        key.set_contents_from_filename(tarball_path, cb=percent_callback, num_cb=40)
+
+        public_url = 'https://{bucket}.{host}/{key}'.format(
+            host=conn.server_name(),
+            bucket=bucket.name,
+            key=key.key)
+        print()
+        print("Uploaded to %s" % (public_url,))
```

Comment:
So this whole block of code should just be part of the `UploadBackend` classes in `zerver/lib/upload.py`, without the print statements (which should be the same for both code paths).  So this block of code should just be:

```
print("Uploading export tarball")
public_url = upload_backend.upload_export_tarball(...)
print()
print("Uploaded to %s" % (public_url,))
```

with two implementations in the `upload_backend` classes, just like everything else we do for uploading stuff.